src/OperationalInsights/OperationalInsights/LinkedStorageAccount/SetAzureOperationalInsightsLinkedStorageAccountCommand.cs

```diff
@@ -38,9 +38,9 @@ public class SetAzureOperationalInsightsLinkedStorageAccountCommand : Operationa
 
         [Parameter(Position = 2,
             Mandatory = true,
-            HelpMessage = "Data Source Type should be one of 'CustomLogs', 'AzureWatson'.")]
+            HelpMessage = "Data Source Type should be one of 'CustomLogs', 'AzureWatson', 'Query', 'Alerts'.")]
         [ValidateNotNullOrEmpty]
-        [ValidateSet("CustomLogs", "AzureWatson")]
+        [ValidateSet("CustomLogs", "AzureWatson", "Query", "Alerts", IgnoreCase = true)]
```

Comment:
I don't know how frequently the values of DataSourceType change, but if you don't want to block customers to use your latest type, you can change `ValidateSet` to `ArgumentCompleter` which provides the same tab-completion feature but does not validate the argument.

Reply:
these are ARM resources types, it can be left as is